OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationGenerationJob.java

```diff
@@ -40,258 +40,266 @@
+         *    a 30 second timer and attempt to bubble to the AppNotificationWillShowInForegroundHandler automatically
+         */
+        public synchronized void complete(boolean bubble) {
+            destroyTimeout();
+
+            if (isComplete)
+                return;
+
+            isComplete = true;
+
+            // Move on to showing notification if no AppNotificationWillShowInForegroundHandler exi
+            //    or bubbling is set false
+            if (OneSignal.appNotificationWillShowInForegroundHandler == null || !bubble) {
+                GenerateNotification.fromJsonPayload(getNotifJob());
+                return;
+            }
+
+            // If the appNotificationWillShowInForegroundHandler exists and we want to bubble, call
+            //    the notificationWillShowInForeground implementation
+            OneSignal.appNotificationWillShowInForegroundHandler.notificationWillShowInForeground(
+                    getNotifJob().toAppNotificationGenerationJob());
+        }
+    }
 
    /**
     * Used to modify the {@link OSNotificationGenerationJob} inside of the {@link OneSignal.AppNotificationWillShowInForegroundHandler}
     *    without exposing internals publicly
     */
    public static class AppNotificationGenerationJob extends NotificationGenerationJob {
 
-      AppNotificationGenerationJob(OSNotificationGenerationJob notifJob) {
-         super(notifJob);
+        AppNotificationGenerationJob(OSNotificationGenerationJob notifJob) {
+            super(notifJob);
 
-         startTimeout(new Runnable() {
-            @Override
-            public void run() {
-               AppNotificationGenerationJob.this.complete();
-            }
-         });
-      }
 
-      // Method controlling completion from the AppNotificationWillShowInForegroundHandler
-      //    If a dev does not call this at the end of the notificationWillShowInForeground implementation, a runnable will fire after
-      //    a 30 second timer and complete by default
-      public synchronized void complete() {
-         destroyTimeout();
+            startTimeout(new Runnable() {
+                @Override
+                public void run() {
+                    AppNotificationGenerationJob.this.complete();
+                }
+            });
+        }
 
-         if (isComplete)
-            return;
+        /**
+         * Method controlling completion from the AppNotificationWillShowInForegroundHandler
+         *    If a dev does not call this at the end of the notificationWillShowInForeground implementation, a runnable will fire after
+         *    a 30 second timer and complete by default
+         */
+        public synchronized void complete() {
+            destroyTimeout();
 
-         isComplete = true;
+            if (isComplete)
+                return;
 
-         GenerateNotification.fromJsonPayload(getNotifJob());
-      }
-   }
+            isComplete = true;
+
+            //
```

Comment:
Remove blank comment

Reply:
I actually should add a comment for this haha

Reply:
Fixed!